src/ucai/core/client.py

```diff
@@ -43,6 +43,10 @@ def __init__(self, client: Any = None, **kwargs: Any) -> None:  # noqa: ARG002
     def create_function(self, *args: Any, **kwargs: Any) -> Any:
         """Create a function"""
 
+    @abstractmethod
+    def create_python_function(self, *, func: Callable[..., Any], func_comment: str, catalog: str, schema: str) -> Any:
```

Comment:
Is `func_comment` required?

Reply:
yeah, without it, the GenAI tools just sort of "guess" at what a tool is. In tutorials, we'll be making this VERY clear that these are needed to make high-quality agentic tools

Reply:
Can we extract if from the docstring?

Reply:
After doing some bug bashing, this makes a ton of sense. It's really annoying to type out a description if I already have it in my docstring. 
I have:
- Raised a ValueError if the description isn't supplied in the docstring
- Remove the argument since it's redundant
- Added tests to confirm the behavior of not supplying / providing an empty string as the description raises a ValueError